mono/utils/mono-time.c

```diff
@@ -161,6 +178,31 @@ mono_100ns_ticks (void)
 }
 
 /*
+ * Returns user + system time for a thread in 100ns units.
+ */
+gint64
+mono_thread_cpu_time (void)
+{
+#if defined(__linux__)
+	struct timespec ts;
+	clock_gettime (CLOCK_THREAD_CPUTIME_ID, &ts);
+	return ts.tv_sec * 10000000 + ts.tv_nsec / 100;
+#elif defined(PLATFORM_MACOSX)
+	mach_msg_type_number_t count = THREAD_BASIC_INFO_COUNT;
+	thread_basic_info_data_t info;
+
+	if (thread_info (pthread_mach_thread_np (pthread_self ()), THREAD_BASIC_INFO,
```

Comment:
It's not pretty much everything we do, might have been the case in the past, but it's time we learn that lesson. We did pipe up hacks around platform details and they come to bite us all the time. So, let's avoid this trap whenever possible.

We should stick to well documented and defined parts of each platform.